### PR TITLE
Add easy way to serialize CUDS objects

### LINKED ISSUE #322: Add easy way to serialize CUDS objects
In GitLab by @urbanmatthias on Jun 8, 2020, 17:42

- update `__getstate__`
- implemet a `to_json` method
- add a method to utils

### PR DESCRIPTION
**Pull request by urbanmatthias**
Fixes #322

### CODE REVIEW

**Comment by yoavnash** [comment]
How is it now easier?

**Comment by urbanmatthias** [comment]
There are deserialize and serialize functions in osp.core.utils, which can be called with a cuds objects only.

No need to specify a session, or call find_cuds_object. Although it is still a allowed to specify a session, if the user wants to.

File: tests/test_utils.py [hunk 1]
```diff
@@ -201,6 +217,31 @@ def test_deserialize(self):
...
+
+    def test_serialize(self):
+        c = branch(
+            city.City(name="Freiburg", uid=1),
+            branch(
+                city.Neighborhood(name="Littenweiler", uid=2),
+                city.Street(name="Schwarzwaldstraße", uid=3)
+            )
+        )
+        self.maxDiff = None
```
Comment (yoavnash) [inline tests/test_utils.py:236]: What does this do?

Comment (urbanmatthias) [reply tests/test_utils.py:236]: The self.maxDiff? In case there is an error in the unittest, it will print the entire diff and not only one line

**Comment by yoavnash** [comment]
Alright, great. Then I think this together with the openAPI document should be enough for now, right?

**Comment by urbanmatthias** [comment]
I hope so

**Pull request merged**